tests/ui/typeck/issue-98982.stderr

```diff
@@ -8,26 +8,11 @@ LL | |         return i;
 LL | |     }
    | |_____^ expected `i32`, found `()`
    |
-note: the function expects a value to always be returned, but loops might run zero times
-  --> $DIR/issue-98982.rs:2:5
-   |
-LL |     for i in 0..0 {
-   |     ^^^^^^^^^^^^^ this might have zero elements to iterate on
-LL |         return i;
-   |         -------- if the loop doesn't execute, this value would never get returned
-help: return a value for the case when the loop has zero elements to iterate on
+help: consider returning a value here
    |
 LL ~     }
 LL ~     /* `i32` value */
    |
-help: otherwise consider changing the return type to account for that possibility
-   |
-LL ~ fn foo() -> Option<i32> {
-LL |     for i in 0..0 {
-LL ~         return Some(i);
-LL ~     }
-LL ~     None
-   |
```

Comment:
I don't think that this reversal is reasonable, because the explanation *for this case* is exactly right: `for i in 0..0` will never iterate, right?

Reply:
Yes, `for i in 0..0` will never iterate, but that is not what is causing the error. `for i in 0..100` will definitely iterate but the same error still appears.